text/3874-build-std-always.md

```diff
@@ -0,0 +1,1538 @@
+
+## Why invert the `mem` feature?
+[rationale-no-mem]: #why-invert-the-mem-feature
+
+While "negative" features are typically discouraged due to how features unify
+(e.g. `std` features are preferred to `no_std`): the `mem` feature's current
+behaviour is the opposite of what is optimal.
+
+Ideally, a crate should be able to provide alternate memory symbols and disable
+`compiler_builtins`' symbols for the entire crate graph by enabling a feature
+(e.g. `std`/`libc` could do this) - this is what an `external-mem` feature
+enables.
+
+↩ [*`compiler-builtins-mem`*][compiler-builtins-mem]
+
+## Why not use weak linkage for `compiler-builtins/mem` symbols?
+[rationale-no-weak-linkage]: #why-not-use-weak-linkage-for-compiler-builtinsmem-symbols
+
+Since [compiler-builtins#411], the relevant symbols in `compiler_builtins`
+already have weak linkage. However, it is nevertheless not possible to simply
+remove the `mem` feature and have the symbols always be present:
+
+- Some targets, such as those based on MinGW, do not have sufficient support for
+  weak definitions (at least with the default linker).
+- Weak linkage has precedence over shared libraries and the symbols of a
+  dynamically-linked `libc` should be preferred over `compiler_builtins`'s
+  symbols.
+
+↩ [*`compiler-builtins-mem`*][compiler-builtins-mem]
+
+## Why not globally cache builds of the standard library?
+[rationale-caching]: #why-not-globally-cache-builds-of-the-standard-library
+
+The standard library is no different than regular dependencies in being able to
+benefit from global caching of dependency builds. It is out-of-scope of this
+proposal to propose a special-cased mechanism for this that applies only to the
+standard library. [cargo#5931] tracks the feature request of intermediate
+artifact caching in Cargo.
+
+↩ [*Caching*][caching]
+
+## Why not link to hosted standard library documentation in generated docs?
+[rationale-generated-docs]: #why-not-link-to-hosted-standard-library-documentation-in-generated-docs
+
+Cargo would need to pass `-Zcrate-attr="doc(html_root_url=..)"` to the standard
+library crates when building them but doesn't have the required information to
+know what url to provide. Cargo would require knowledge of the current toolchain
+channel to build the correct url and doesn't know this.
+
+↩ [*Generated documentation*][generated-documentation]
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+The following are aspects of the proposal which warrant further discussion or
+small details are likely to be bikeshed prior to this part of the RFC's
+acceptance or stabilisation and aren't pertinent to the overall design:
+
+## What should the `build-std` configuration in `.cargo/config` be named?
+[unresolved-config-name]: #what-should-the-build-std-configuration-in-cargoconfig-be-named
+
+What should this configuration option be named? `build-std`?
+`rebuild-standard-library`?
```

Comment:
I thought we had notes on considerations for this?

For instance, `build-std = "always"` could be misinterpreted as "don't apply rebuild rules"